sys/random/seed.c

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+ /**
+ * @ingroup sys_random
+ * @{
+ * @file
+ *
+ * @brief PRNG seeding
+ *
+ * @author Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "log.h"
+#include "luid.h"
+#include "random.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+void auto_init_random(void)
+{
+    uint32_t seed;
+#ifdef FEATURE_PERIPH_CPUID
+    luid_get(&seed, 4);
+#else
```

Comment:
There is no need for this else branch: `luid` does already do this, so why do it here again?

Reply:
In case there is no CPUID, we want to seed differently and not by the LUID default value.

Reply:
hm, but why? Where is the difference in seeding with a 0 or with just using the LUID default value?

Alternatively: why not use `luid_custom()` then? This way you get a CPUID based ID if cpuid is available, and something different than the `luid` default value in any case...

Reply:
Seeding with 0 is not intended. Providing an alternative will be part of an other PR. `luid_custom` is not sufficient because we need an other source which is (hopefully) board unique and static between re-boots.

Reply:
ah, I think I see what you mean, so the `else` path is basically only a placeholder here, right?!

Reply:
AFAIR I kept that in so that in case there is no cpuid, the dependency of random to luid can be dropped altogether.

Reply:
I see. So never mind (although this seems to point in the direction of the famous 'premature optimization', as `luid` is neglectably if no `cpuid` module is present?!)